app/celery/tasks.py

```diff
@@ -420,22 +420,23 @@ def update_letter_notifications_statuses(self, filename):
             update_letter_notification(filename, temporary_failures, update)
             sorted_letter_counts[update.cost_threshold] += 1
 
-        if temporary_failures:
-            # This will alert Notify that DVLA was unable to deliver the letters, we need to investigate
-            message = "DVLA response file: {filename} has failed letters with notification.reference {failures}".format(
-                filename=filename, failures=temporary_failures)
-            raise DVLAException(message)
-
-        if sorted_letter_counts.keys() - {'Unsorted', 'Sorted'}:
-            unknown_status = sorted_letter_counts.keys() - {'Unsorted', 'Sorted'}
-
-            message = 'DVLA response file: {} contains unknown Sorted status {}'.format(
-                filename, unknown_status
-            )
-            raise DVLAException(message)
-
-        billing_date = get_billing_date_in_bst_from_filename(filename)
-        persist_daily_sorted_letter_counts(billing_date, sorted_letter_counts)
+        try:
+            if sorted_letter_counts.keys() - {'Unsorted', 'Sorted'}:
+                unknown_status = sorted_letter_counts.keys() - {'Unsorted', 'Sorted'}
+
+                message = 'DVLA response file: {} contains unknown Sorted status {}'.format(
+                    filename, unknown_status
+                )
+                raise DVLAException(message)
+
+            billing_date = get_billing_date_in_bst_from_filename(filename)
+            persist_daily_sorted_letter_counts(billing_date, sorted_letter_counts)
+        finally:
+            if temporary_failures:
+                # This will alert Notify that DVLA was unable to deliver the letters, we need to investigate
+                message = "DVLA response file: {filename} has failed letters with notification.reference {failures}" \
+                    .format(filename=filename, failures=temporary_failures)
+                raise DVLAException(message)
 
 
 def get_billing_date_in_bst_from_filename(filename):
```

Comment:
do we need to split the filename on "-"

Reply:
Yes, we definitely do - I had forgotten to add that bit back in when I changed the PR.